src/Version/Version.cpp

```diff
@@ -118,13 +115,18 @@ std::string getOnnxMlirFullVersion(bool toIncludeLLVM) {
 #endif
 
 void getVersionPrinter(llvm::raw_ostream &os) {
+#if defined(ONNX_MLIR_VENDOR) && defined(ONNX_MLIR_PRODUCT_VERSION)
+  os << ONNX_MLIR_VENDOR << " " << PRODUCT_VERSION_MAJOR << '.';
+  os << PRODUCT_VERSION_MINOR << '.';
+  os << PRODUCT_VERSION_PATCH << '-' << PRODUCT_ID << '\n';
```

Comment:
Aren't `PRODUCT_VERSION_MAJOR`,  `PRODUCT_VERSION_MINOR`, and `PRODUCT_VERSION_PATCH` and `PRODUCT_ID` built by splitting apart the `ONNX_MLIR_PRODUCT_VERSION` supplied at compile time? If so, is there a reason to not use `ONNX_MLIR_PRODUCT_VERSION` directly instead of rebuilding it from the composite components here?

Reply:
Yes there is a reason... if I attempt to use `ONNX_MLIR_PRODUCT_VERSION` it will fail on the `cmake` command with the following error:

```
on.dir/Version.cpp.o -MF src/Version/CMakeFiles/OMVersion.dir/Version.cpp.o.d -o src/Version/CMakeFiles/OMVersion.dir/Version.cpp.o -c ../src/Version/Version.cpp
<command-line>: error: too many decimal points in number
../src/Version/Version.cpp:122:16: note: in expansion of macro 'ONNX_MLIR_PRODUCT_VERSION'
  122 |   os << " " << ONNX_MLIR_PRODUCT_VERSION << '\n';
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~
../src/Version/Version.cpp: In function 'void onnx_mlir::getVersionPrinter(llvm::raw_ostream&)':
<command-line>: error: 'zdlc' was not declared in this scope
<command-line>: note: in definition of macro 'ONNX_MLIR_PRODUCT_VERSION'
[452/509] Building CXX object test/num...CMakeFiles/TestLoop.dir/TestLoop.cpp.o
ninja: build stopped: subcommand failed.

```

I tried 3 times....so adding the code the way I did gets around this problem :)

Reply:
Also I am not rebuilding it.... ONNX_MLIR_PRODUCT_VERSION is broken up into these 4 segments by using a `CMAKE_MATCH_1` and then a SET which was already added to the code.

Reply:
Ok. I thought that might be the case but I wanted to make sure. PR looks good to me. I'd say give it overnight to see if there's any other comments from other timezones but otherwise it's good to go.